nixos/tests/c2FmZQ.nix

```diff
@@ -0,0 +1,77 @@
+import ./make-test-python.nix ({ pkgs, ... }: {
+  name = "c2FmZQ";
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ hmenke ];
+  };
+
+  nodes.machine = {
+    services.c2FmZQ-server = {
+      enable = true;
+      port = 8080;
+      passphraseFile = builtins.toFile "pwfile" "hunter2"; # don't do this on real deployments
+      settings = {
+        verbose = 3; # debug
+      };
+    };
+    environment = {
+      sessionVariables = {
+        C2FMZQ_PASSPHRASE = "lol";
+        C2FMZQ_API_SERVER = "http://localhost:8080";
+      };
+      systemPackages = [
+        pkgs.c2FmZQ
+        (pkgs.writeScriptBin "c2FmZQ-client-wrapper" ''
+          #!${pkgs.expect}/bin/expect -f
+          spawn c2FmZQ-client {*}$argv
+          expect {
+            "Enter password:" { send "$env(PASSWORD)\r" }
+            "Type YES to confirm:" { send "YES\r" }
+            timeout { exit 1 }
+            eof { exit 0 }
+          }
+          interact
+        '')
+      ];
+    };
+  };
+
+  testScript = { nodes, ... }: ''
+    machine.start()
+    machine.wait_for_unit("c2FmZQ-server.service")
+    machine.wait_for_open_port(8080)
+
+    # Create accounts for alice and bob
+    machine.succeed("PASSWORD=foobar c2FmZQ-client-wrapper -- -v 3 create-account alice@example.com")
+    machine.succeed("PASSWORD=fizzbuzz c2FmZQ-client-wrapper -- -v 3 create-account bob@example.com")
+
+    # Log in as alice
+    machine.succeed("PASSWORD=foobar c2FmZQ-client-wrapper -- -v 3 login alice@example.com")
+    msg = machine.succeed("c2FmZQ-client -v 3 status")
+    assert "Logged in as alice@example.com" in msg, f"ERROR: Not logged in as alice:\n{msg}"
+
+    # Create a new album, upload a file, and delete the uploaded file
+    machine.succeed("c2FmZQ-client -v 3 create-album 'Rarest Memes'")
+    machine.succeed("echo 'pls do not steal' > meme.txt")
+    machine.succeed("c2FmZQ-client -v 3 import meme.txt 'Rarest Memes'")
+    machine.succeed("c2FmZQ-client -v 3 sync")
+    machine.succeed("rm meme.txt")
+
+    # Share the album with bob
+    machine.succeed("c2FmZQ-client-wrapper -- -v 3 share 'Rarest Memes' bob@example.com")
+
+    # Log in as bob
+    machine.succeed("PASSWORD=fizzbuzz c2FmZQ-client-wrapper -- -v 3 login bob@example.com")
+    msg = machine.succeed("c2FmZQ-client -v 3 status")
+    assert "Logged in as bob@example.com" in msg, f"ERROR: Not logged in as bob:\n{msg}"
+
+    # Download the shared file
+    machine.succeed("c2FmZQ-client -v 3 download 'shared/Rarest Memes/meme.txt'")
+    machine.succeed("c2FmZQ-client -v 3 export 'shared/Rarest Memes/meme.txt' .")
+    msg = machine.succeed("cat meme.txt")
+    assert "pls do not steal\n" == msg, f"File content is not the same:\n{msg}"
+
+    # Test that PWA is served
+    msg = machine.succeed("curl -sSfL http://localhost:8080")
+    assert "c2FmZQ" in msg, f"Could not find 'c2FmZQ' in the output:\n{msg}"
```

Comment:
I think it would be better to use `with subtest("Download the shared file"):`, etc. to label these logical blocks, instead of inlining comments. That way, you both get the logical blocks, and the test will tell you which logical block if failed in.

Maybe the uppermost ones which don't test the service, rather just set up stuff doesn't need to be marked as subtests. I don't know.